src/pe_reports/report_generator.py

```diff
@@ -28,39 +21,223 @@
+    filesize = os.path.getsize(output)
+    if filesize >= 20000000:
+        tooLarge = True
+
+    return filesize, tooLarge
+
+
+def convert_html_to_pdf(source_html, output_filename):
+    """Convert html to pdf."""
+    # Open output file for writing (truncated binary)
+    result_file = open(output_filename, "w+b")
+
+    # Convert HTML to PDF
+    pisa_status = pisa.CreatePDF(
+        source_html, dest=result_file  # the HTML to convert
+    )  # file handle to receive result
+
+    # Close output file
+    result_file.close()  # close output file
+
+    # Return False on success and True on errors
+    return pisa_status.err
+
+
+def generate_reports(datestring, output_directory):
+    """Process steps for generating report data."""
+    # Get PE orgs from PE db
+    conn = connect()
+    if conn:
+        pe_orgs = get_orgs(conn)
+    else:
+        return 1
+    generated_reports = 0
+
+    # Iterate over organizations
+    if pe_orgs:
+        logging.info(f"PE orgs count: {len(pe_orgs)}")
+        for org in pe_orgs:
+            # Assign organization values
+            org_uid = org[0]
+            org_name = org[1]
+            org_code = org[2]
+
+            logging.info(f"Running on {org_code}")
 
+            # Create folders in output directory
+            for dir_name in ("ppt", org_code):
+                if not os.path.exists(f"{output_directory}/{dir_name}"):
+                    os.mkdir(f"{output_directory}/{dir_name}")
 
-def generate_reports(data, data_dir, out_dir, db_creds_file):
-    """Gather assets to produce reports."""
+            # Load source HTML
+            try:
+                basedir = os.path.abspath(os.path.dirname(__file__))
+                template = os.path.join(basedir, "template.html")
+                file = open(template)
+                source_html = file.read().replace("\n", " ")
+            except FileNotFoundError:
+                logging.error(
+                    "Template cannot be found. It must be named: '%s'", template
+                )
+                return 1
 
+            # Insert Charts and Metrics into PDF
+            (
+                source_html,
+                hibp_creds,
+                cyber_creds,
+                masq_df,
+                insecure_df,
+                vulns_df,
+                assets_df,
+                dark_web_mentions,
+                alerts,
+                top_cves,
+            ) = init(
+                source_html,
+                datestring,
+                org_name,
+                org_uid,
+            )
 
-def main() -> None:
-    """Set up logging and call the generate_reports function."""
+            # Close PDF
+            file.close()
+
+            # Convert to HTML to PDF
+            output_filename = f"{output_directory}/{org_code}-Posture_and_Exposure_Report-{datestring}.pdf"
+            convert_html_to_pdf(source_html, output_filename)
+
+            # Create Crendential Exposure Excel file
```

Comment:
Just noticed this typo:
```suggestion
            # Create Credential Exposure Excel file
```